File: dispatch/java/adamma/c4dhi/claid/Module/Scheduling/FunctionRunnable.java
package adamma.c4dhi.claid.Module.Scheduling;

import java.util.function.Consumer;

public class FunctionRunnable extends ScheduledRunnable
{

    final Runnable runnable;

    public FunctionRunnable(final Runnable runnable)
    {
        super(ScheduleOnce.now());
        this.runnable = runnable;
    }

    public FunctionRunnable(final Runnable runnable, ScheduleDescription schedule)
    {
        super(schedule);
        this.runnable = runnable;
    }

    @Override
    public void run() {
        runnable.run();
    }    
}

File: dispatch/ios/CLAID/Sources/CLAID/Module/Scheduling/FunctionRunnable.swift
//
//  FunctionRunnable.swift
//  CLAID
//
//  Created by Patrick Langer on 20.02.2025.
//


public final actor FunctionRunnable: ScheduledRunnable {
    
    public let schedule: ScheduleDescription
    private let runnable: @Sendable () async -> Void
    private var valid: Bool = true
    
    public func isValid() async -> Bool {
        return valid
    }
    
    public  init(schedule: ScheduleDescription, runnable: @Sendable @escaping () async -> Void) {
        self.schedule = schedule
        self.runnable = runnable
    }
    

    public func run() async {
        guard await isValid() else { return }
        await runnable()
    }
    
    public func invalidate() async {
        valid = false
    }
}
